src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/WindowsRuntime/Windows/UI/ViewManagement/InputPaneRcw.cs

```diff
@@ -19,6 +19,36 @@ namespace Windows.UI.ViewManagement
         /// </summary>
         internal static class InputPaneRcw
         {
+            [DllImport("api-ms-win-core-winrt-string-l1-1-0.dll", CallingConvention = CallingConvention.StdCall)]
```

Comment:
DllImport definitions should be moved over to common files. They are named xxxNativeMethods.cs.

Reply:
does this require deploying `api-ms-win-core-winrt-string-l1-1-0.dll` for Win7 (which is still supported as target by .NET Core) ?

Reply:
In Windows 7, the WPF InputPane wrapper classes are irrelevant.  WPF touch keyboard support works in Windows 7 without any intervention.  When focus tracking was changed to use WM_POINTER in Win8+, WPF touch keyboard support broke because WISP (which underlies all WPF touch/stylus support) redirects input messages to a separate thread and does not allow WM_POINTER* messages to flow to the actual message loop.  This was added in response to that change (although much later) so that WPF could explicitly request the touch keyboard to show.

Reply:
In terms of Windows 10, the APIs needed are not only Win10 only, but 1607+:
https://docs.microsoft.com/en-us/windows/win32/api/inputpaneinterop/nn-inputpaneinterop-iinputpaneinterop

They were added (initially) to support this WPF mechanism, though they're useful for all Win32 applications.

Reply:
I was talking about the DLL reference to `api-ms-win-core-winrt-string-l1-1-0.dll` on Windows 8.1 (or whatever the version check will let through) resolving correctly in general, not about the particular InputPaneRCW

Reply:
Right, I am saying a DLLNotFoundException is acceptable on Win8.1 as the platform cannot support the InputPane usage anyway.  We'll have to ensure that exception is caught though (@jkoritzinsky) and appropriately handled to mark the InputPane mechanisms as not supported.

Reply:
I'm updating the initialization to handle the `DllNotFoundException` and have the code go down the same "mark as unsupported" path.